### ISSUE TITLE
Concurrent deletion of about 70+ containers failed(from Management UI)

### ISSUE THREAD

**Issue by lgayatri** [vmware/vic#5157]
Concurrent deletion of about 70+ containers failed with below error:

request failed, no additional details provided; Reason: {"message":"request failed, no additional details provided","stackTrace":["com.vmware.xenon.common.Operation.fail(Operation.java:1227)","com.vmware.xenon.common.http.netty.NettyHttpServiceClient.failExpiredRequests(NettyHttpServiceClient.java:834)","com.vmware.xenon.common.http.netty.NettyHttpServiceClient.handleMaintenance(NettyHttpServiceClient.java:781)","com.vmware.admiral.adapter.docker.service.RemoteApiDockerAdapterCommandExecutorImpl.handleMaintenance(RemoteApiDockerAdapterCommandExecutorImpl.java:515)","com.vmware.admiral.adapter.docker.service.AbstractDockerAdapterService.handleMaintenance(AbstractDockerAdapterService.java:102)","com.vmware.xenon.common.ServiceMaintenanceTracker.lambda$performServiceMaintenance$1(ServiceMaintenanceTracker.java:171)","com.vmware.xenon.common.ServiceHost.executeRunnableSafe(ServiceHost.java:5019)","com.vmware.xenon.common.ServiceHost.lambda$schedule$55(ServiceHost.java:5013)","java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)","java.util.concurrent.FutureTask.run(FutureTask.java:266)","java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)","java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)","java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)","java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)","java.lang.Thread.run(Thread.java:745)"],"statusCode":408,"documentKind":"com:vmware:xenon:common:ServiceErrorResponse","errorCode":0}

Attached log bundle
[vm-2017-05-18.6905.tar.gz](https://github.com/vmware/vic/files/1010804/vm-2017-05-18.6905.tar.gz)

**Comment by mdubya66** [comment]
Should this be opened in admiral?

**Comment by jzt** [comment]
@jooskim can you take a gander at this and let me know if this belongs as a UI-related bug on our end?

**Comment by jooskim** [comment]
```
"com.vmware.admiral.adapter.docker.service.RemoteApiDockerAdapterCommandExecutorImpl.handleMaintenance(RemoteApiDockerAdapterCommandExecutorImpl.java:515)"
```
seems like an admiral issue

**Comment by mdubya66** [comment]
it's also a VIC issue

She has 50 VCH's. I pulled the logs from one of them 
[container-logs (1).zip](https://github.com/vmware/vic/files/1011458/container-logs.1.zip)
docker ps shows starting for 12 hours
```
(mwilliamson) ib-aus-office-164:~/Downloads/vic 3 $ docker -H 10.197.37.134:2376 --tls ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
e7704f0b632d        busybox             "top"               12 hours ago        Starting                                trusting_banach
f0bc3b4d2fc5        busybox             "top"               12 hours ago        Starting                                thirsty_edison
77c54aecd271        busybox             "top"               12 hours ago        Starting                                hardcore_babbage
65520884ff73        busybox             "top"               12 hours ago        Starting                                silly_curran
185fa596ec7a        busybox             "top"               12 hours ago        Starting                                epic_albattani
f921d5bdff2b        busybox             "top"               12 hours ago        Starting                                blissful_torvalds
8778a0cfd86f        busybox             "top"               12 hours ago        Starting                                brave_stallman
509ab79a0da2        busybox             "top"               12 hours ago        Starting                                condescending_bardeen
71aeb191c494        busybox             "top"               12 hours ago        Starting                                unruffled_mccarthy
2e30dd2c21d4        busybox             "top"               12 hours ago        Starting                                flamboyant_kare
4041cf624948        busybox             "top"               12 hours ago        Starting                                agitated_cori
f85383d887e2        busybox             "top"               12 hours ago        Starting                                festive_bohr
e7a9b6ece9f6        busybox             "top"               12 hours ago        Starting                                kickass_albattani
876d1377620f        busybox             "top"               12 hours ago        Starting                                dreamy_cray
75341b9885d7        busybox             "top"               12 hours ago        Up 4 hours                              cranky_boyd
89f2e1207f0a        busybox             "top"               12 hours ago        Up 4 hours                              gracious_kalam
48d4e51d7d9d        busybox             "top"               12 hours ago        Up 4 hours                              priceless_yonath
ec00e2e0709c        busybox             "top"               12 hours ago        Up 4 hours                              wonderful_mccarthy
1ade3128aaf7        busybox             "top"               12 hours ago        Up 5 hours                              cranky_wozniak
76a6297b48a5        busybox             "top"               12 hours ago        Up 5 hours                              focused_aryabhata
f5613dd80dda        busybox             "top"               12 hours ago        Up 5 hours                              upbeat_shockley
941678b022c5        busybox             "top"               12 hours ago        Up 5 hours                              angry_ardinghelli
be61094779b3        busybox             "top"               12 hours ago        Up 5 hours                              zealous_blackwell
0c911139571f        busybox             "top"               12 hours ago        Up 5 hours                              practical_heyrovsky
accccdef4813        busybox             "top"               12 hours ago        Up 5 hours                              distracted_bell
0a93e4259b1c        busybox             "top"               12 hours ago        Up 5 hours                              mystifying_wescoff
ab5d48c7bdfc        busybox             "top"               12 hours ago        Up 5 hours                              peaceful_easley
6d4e81e84e83        busybox             "top"               12 hours ago        Up 5 hours                              stoic_albattani
bdf87ecaeb05        busybox             "top"               12 hours ago        Up 5 hours                              pensive_bartik
56d8cac81adf        busybox             "top"               12 hours ago        Up 5 hours                              optimistic_lovelace
2f21c6664c23        busybox             "top"               12 hours ago        Up 5 hours                              keen_mccarthy
62078cb7e143        busybox             "top"               12 hours ago        Up 5 hours                              sleepy_spence
b1d3d51c4ba1        busybox             "top"               12 hours ago        Up 5 hours                              amazing_wright
3538560a18ae        busybox             "top"               12 hours ago        Up 5 hours                              elegant_mcclintock
032519e1e517        busybox             "top"               12 hours ago        Up 5 hours                              festive_shannon
5b1fc9d1f556        busybox             "top"               12 hours ago        Up 5 hours                              affectionate_payne
```
docker stats looks weird
```CONTAINER           CPU %               MEM USAGE / LIMIT   MEM %               NET I/O             BLOCK I/O           PIDS
75341b9885d7        --                  -- / --             --                  --                  --                  --
89f2e1207f0a        --                  -- / --             --                  --                  --                  --
48d4e51d7d9d        --                  -- / --             --                  --                  --                  --
ec00e2e0709c        --                  -- / --             --                  --                  --                  --
1ade3128aaf7        --                  -- / --             --                  --                  --                  --
76a6297b48a5        --                  -- / --             --                  --                  --                  --
f5613dd80dda        --                  -- / --             --                  --                  --                  --
941678b022c5        --                  -- / --             --                  --                  --                  --
be61094779b3        --                  -- / --             --                  --                  --                  --
0c911139571f        --                  -- / --             --                  --                  --                  --
accccdef4813        --                  -- / --             --                  --                  --                  --
0a93e4259b1c        --                  -- / --             --                  --                  --                  --
ab5d48c7bdfc        --                  -- / --             --                  --                  --                  --
6d4e81e84e83        --                  -- / --             --                  --                  --                  --
bdf87ecaeb05        --                  -- / --             --                  --                  --                  --
56d8cac81adf        --                  -- / --             --                  --                  --                  --
2f21c6664c23        --                  -- / --             --                  --                  --                  --
62078cb7e143        --                  -- / --             --                  --                  --                  --
b1d3d51c4ba1        --                  -- / --             --                  --                  --                  --
3538560a18ae        --                  -- / --             --                  --                  --                  --
5b1fc9d1f556        --                  -- / --             --                  --                  --                  --
032519e1e517        --                  -- / --             --                  --                  --                  --
```

**Comment by mdubya66** [comment]
The port-layer log has
```
May 18 2017 08:41:09.796Z DEBUG [ END ] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).collect:169] [56.473364ms] 
May 18 2017 08:41:09.796Z DEBUG container stats complete for 3538560a18ae787e094df3204cd00104c28ea858b149c75a0be1fe7cc1f2d758 @ 842350520512
May 18 2017 08:41:09.796Z DEBUG Error streaming containerStats for be61094779b3fc10795f11532dfc95746bddf5d13c202a4e39597ced8f3bb85c: io: read/write on closed pipe
May 18 2017 08:41:09.803Z DEBUG Error streaming containerStats for 76a6297b48a5441a71c46bf573188c04d7e3ffa432b2ff1817443e295f2bbb0f: io: read/write on closed pipe
May 18 2017 08:41:09.804Z DEBUG Error streaming containerStats for 6d4e81e84e83e222decfc2f31548e21de2e04eaced53db21858c8db3743c0579: io: read/write on closed pipe
May 18 2017 08:41:09.818Z DEBUG Error streaming containerStats for 0c911139571f0b72c04cca32975ee61d7cd35c3f0c1023101f3bb2d542212d13: io: read/write on closed pipe
May 18 2017 08:41:10.761Z DEBUG The VCH stats are: [-1 -1 2392 9470738432]
May 18 2017 08:41:10.763Z DEBUG The VCH stats are: [-1 -1 2392 9470738432]
May 18 2017 08:41:10.788Z DEBUG The VCH stats are: [-1 -1 2392 9470738432]
May 18 2017 08:41:10.789Z DEBUG The VCH stats are: [230334 1724971 125788 521262858240]
May 18 2017 08:41:10.790Z DEBUG [BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerStatsHandler:327] 56d8cac81adf2127daa61f24b0c10d3c087a7c330de42f586a1c6c973f35394e
May 18 2017 08:41:10.790Z DEBUG [BEGIN] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).Sample:146]
May 18 2017 08:41:10.790Z DEBUG [ END ] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).Sample:146] [39.546µs] 
May 18 2017 08:41:10.790Z DEBUG container(56d8cac81adf2127daa61f24b0c10d3c087a7c330de42f586a1c6c973f35394e) stats sample subscribed @ 842350521184
May 18 2017 08:41:10.790Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerStatsHandler:327] [126.36µs] 56d8cac81adf2127daa61f24b0c10d3c087a7c330de42f586a1c6c973f35394e
May 18 2017 08:41:10.790Z DEBUG [BEGIN] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).collect:169]
May 18 2017 08:41:10.804Z DEBUG The VCH stats are: [230334 1724971 125788 521262858240]
May 18 2017 08:41:10.806Z DEBUG [BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerStatsHandler:327] bdf87ecaeb05c205b3134682b58bf501da79aaccb0d1142d3f63b143b6484523
May 18 2017 08:41:10.806Z DEBUG [BEGIN] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).Sample:146]
May 18 2017 08:41:10.806Z DEBUG [ END ] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).Sample:146] [92.174µs] 
May 18 2017 08:41:10.806Z DEBUG container(bdf87ecaeb05c205b3134682b58bf501da79aaccb0d1142d3f63b143b6484523) stats sample subscribed @ 842350521312
May 18 2017 08:41:10.806Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerStatsHandler:327] [498.499µs] bdf87ecaeb05c205b3134682b58bf501da79aaccb0d1142d3f63b143b6484523
May 18 2017 08:41:10.807Z DEBUG [BEGIN] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).collect:169]
May 18 2017 08:41:10.811Z DEBUG The VCH stats are: [230334 1724971 125788 521262858240]
May 18 2017 08:41:10.812Z DEBUG [BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerStatsHandler:327] accccdef48137a3b6828f13697c6c75836cea3b391705158a80a843608263a4b
May 18 2017 08:41:10.812Z DEBUG [BEGIN] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).Sample:146]
May 18 2017 08:41:10.812Z DEBUG [ END ] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).Sample:146] [94.958µs] 
May 18 2017 08:41:10.812Z DEBUG container(accccdef48137a3b6828f13697c6c75836cea3b391705158a80a843608263a4b) stats sample subscribed @ 842350521416
May 18 2017 08:41:10.813Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerStatsHandler:327] [286.657µs] accccdef48137a3b6828f13697c6c75836cea3b391705158a80a843608263a4b
May 18 2017 08:41:10.813Z DEBUG [BEGIN] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).collect:169]
May 18 2017 08:41:10.827Z DEBUG [ END ] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).collect:169] [36.736788ms] 
May 18 2017 08:41:10.828Z DEBUG container stats complete for 56d8cac81adf2127daa61f24b0c10d3c087a7c330de42f586a1c6c973f35394e @ 842350521176
May 18 2017 08:41:10.837Z DEBUG [ END ] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).collect:169] [30.056449ms] 
May 18 2017 08:41:10.838Z DEBUG [ END ] [github.com/vmware/vic/lib/portlayer/metrics.(*collectorVM).collect:169] [25.808039ms] 
May 18 2017 08:41:10.839Z DEBUG container stats complete for accccdef48137a3b6828f13697c6c75836cea3b391705158a80a843608263a4b @ 842350521408
May 18 2017 08:41:10.839Z DEBUG container stats complete for bdf87ecaeb05c205b3134682b58bf501da79aaccb0d1142d3f63b143b6484523 @ 842350521304
May 18 2017 08:41:11.739Z DEBUG Error streaming containerStats for 3538560a18ae787e094df3204cd00104c28ea858b149c75a0be1fe7cc1f2d758: io: read/write on closed pipe
May 18 2017 08:41:12.791Z DEBUG Error streaming containerStats for 56d8cac81adf2127daa61f24b0c10d3c087a7c330de42f586a1c6c973f35394e: io: read/write on closed pipe
May 18 2017 08:41:12.806Z DEBUG Error streaming containerStats for bdf87ecaeb05c205b3134682b58bf501da79aaccb0d1142d3f63b143b6484523: io: read/write on closed pipe
May 18 2017 08:41:12.813Z DEBUG Error streaming containerStats for accccdef48137a3b6828f13697c6c75836cea3b391705158a80a843608263a4b: io: read/write on closed pipe
```

**Comment by jzt** [comment]
Saw this too:
```
https://10.197.37.142:2376/v1.21/containers/028a671e8034873a0b37cff90f5fc05fe99b64544e43714fe57994520e74d52f?v=1&force=1 returned error 500 for DELETE. id 13266967; Reason: Server error from portlayer: [DELETE /containers/{id}][500] containerRemoveInternalServerError
```

see https://github.com/vmware/vic/issues/1823

**Comment by jzt** [comment]
The logs pasted at the top of this issue show a 408 (timeout) FYI. There are tons of these in the failed request log in admiral as well.

**Comment by mdubya66** [comment]
@https://github.com/vmware/admiral/issues/102 was also filed against admiral

**Comment by anchal-agrawal** [comment]
Marked as medium priority. @mhagen-vmware, please re-assess if needed.

**Comment by corrieb** [comment]
@lgayatri Have you been running this test since this bug was raised? It would be good to know if we're still seeing this issue. It's a significant scalability problem if we're seeing it consistently

**Comment by corrieb** [comment]
@lgayatri ping ^^^^^

**Comment by lgayatri** [comment]
@corrieb , With the OVA build I have - hit : https://github.com/vmware/vic/issues/5737. Currently my setup is down, I will bring it up asap and rerun.

**Comment by lgayatri** [comment]
@corrieb , I moved to newer VC/ESX/NSX builds as my test bed  crashed and now having issues with Logical switch creation on NSX. Once I get it resolved will provide the needinfo

**Comment by lgayatri** [comment]
My Test bed came up partially, and here is the output on build 12202
![image](https://user-images.githubusercontent.com/25006749/28522325-2df7c0da-7095-11e7-89f4-427295b972f0.png)

https://10.197.37.134:2376/v1.19/containers/fb63ef9395b55546df11db37d8a4082d32e34b1f3e0f9a65d9a943325717b83f?v=1&force=1 returned error 500 for DELETE. id 645981; Reason: Server error from portlayer: [DELETE /containers/{id}][500] containerRemoveInternalServerError
[vm-2017-07-24.4067.tar.gz](https://github.com/vmware/vic/files/1169731/vm-2017-07-24.4067.tar.gz)

**Comment by caglar10ur** [comment]
See #5784

**Comment by cgtexmex** [comment]
dupe #5628

**Comment by cgtexmex** [comment]
Might be fixed with #5784 but at a minimum a dupe of #5628
